### PR TITLE
[BugFix] Fix hang because of empty scan ranges

### LINKED ISSUE #9411: Insert data from the Hive external table into the SR internal table is abnormal 【SR 2.3】
Scenario: When the Hive external table is used to write data to the SR internal table, an exception is reported if the data in the Hive external table is empty. The exception may be caused by query timeout. The empty result set is blocked during the INSERT, as shown in the following figure:
image

ps: It is normal to directly query the Hive external table and perform insert into select if there is data in the table

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/StarRocksTest/issues/870

Fixes #9411

## Problem Summary(Required) ：

We don't handle empty scan range properly.

### CODE REVIEW

**Review by imay — APPROVED** [review]
LGTM

File: be/src/exec/vectorized/connector_scan_node.cpp [hunk 1]
```diff
@@ -161,17 +161,21 @@ Status ConnectorScanNode::open(RuntimeState* state) {
...
     RETURN_IF_ERROR(_data_source_provider->open(state));
+
     return Status::OK();
 }
 
 Status ConnectorScanNode::_start_scan_thread(RuntimeState* state) {
     for (TScanRangeParams& scan_range : _scan_ranges) {
         _create_and_init_scanner(state, scan_range.scan_range);
     }
+    _num_scanners = _pending_scanners.size();
```
Comment (DorianZheng) [inline be/src/exec/vectorized/connector_scan_node.cpp:172]: when will `_num_scanners == 0` happen?

Comment (dirtysalt) [reply be/src/exec/vectorized/connector_scan_node.cpp:172]: @DorianZheng when there is no scan file in this table.

Comment (DorianZheng) [reply be/src/exec/vectorized/connector_scan_node.cpp:172]: OK, got it. Why not prevent this case on FE?

**Review by DorianZheng — APPROVED** [review]
left a comment, rest LGTM

File: be/src/exec/vectorized/connector_scan_node.cpp (hunk 1 above)
Comment (dirtysalt) [reply be/src/exec/vectorized/connector_scan_node.cpp:172]: Yes. That would also solve the problem. It generates an EMPTY NODE.

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/exec/vectorized/connector_scan_node.cpp (hunk 1 above)
Comment (DorianZheng) [reply be/src/exec/vectorized/connector_scan_node.cpp:172]: Why does it generate an empty node instead of return early?

**Review by caneGuy — APPROVED** [review]

**Comment by dirtysalt** [comment]
@MergifyIO backport branch-2.3

**Comment by dirtysalt** [comment]
@MergifyIO backport branch-2.4

**Pull request merged**